### PR TITLE
prepare code for new tutorial

### PR DESCRIPTION
**Pull request by leandro-ro**
- removed in-code ganache: Now needed to be called from the command line like in the payment channel tutorial `ganache-cli --host 127.0.0.1 --port 8545 --account 0x50b4 ... `

- removed test environment (main.go & utils.go -> removed main_test.go)  

- removed remainings from collateralized channels (app.go, handle.go, eth.go, util.go)

- removed unused struct (client.go)

### CODE REVIEW

File: app-channel/util.go [hunk 1]
```diff
@@ -55,17 +55,17 @@ func deployContracts(nodeURL string, chainID *big.Int, deploymentKey *ecdsa.Priv
 		return
 	}
 
-	// Deploy collateralized channels app
-	appAddr, txApp, err := ethClient.DeployApp(adjudicatorAddr)
+	// Deploy app
+	appAddr, txApp, err := ethClient.DeployApp()
 	if err != nil {
-		err = errors.WithMessage(err, "deploying CollateralApp")
+		err = errors.WithMessage(err, "deploying TikTakToeApp")
```
Comment (matthiasgeihs) [inline app-channel/util.go:61]:
```suggestion
		err = errors.WithMessage(err, "deploying TikTacToe app")
```

**Review by matthiasgeihs — COMMENTED** [review]
I think we can remove [package perun](https://github.com/perun-network/perun-examples/tree/84ecfec0cc72d5fcbe48cd89842f0a2b629f3e2a/app-channel/perun). We can use `simple.Wallet` instead of the custom wallet implementation and move the client config and setup from [perun.go](https://github.com/perun-network/perun-examples/blob/84ecfec0cc72d5fcbe48cd89842f0a2b629f3e2a/app-channel/perun/perun.go) into package client.

Furthermore, I thought it might be nice  if in the end we can actually run this demo on the command line, like with [perun-eth-demo](https://github.com/perun-network/perun-eth-demo). However, that would need some more changes, so maybe we can postpone it to later.

You can focus on preparing also the new code for the basic payment channel tutorial, which this code here should build upon. Then let me know and we can discuss how to proceed.

File: payment-channel/client/client.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	}
+	ctx, cancel := c.defaultContextWithTimeout()
+	defer cancel()
+
+	// Send the proposal.
+	ch, err := c.PerunClient.StateChClient.ProposeChannel(ctx, proposal)
+	c.Channel = ch
+
+	// Start the on-chain watcher to listen for events
+	c.HandleNewChannel(ch) // TODO: 2/2 Check with MG why this is needed here (and not needed in App Channel example)
```
Comment (leandro-ro) [inline payment-channel/client/client.go:100]: Does the channel-creating client explicitly need to listen for changes in the channel? Or does the creation imply this already? @matthiasgeihs

**Comment by leandro-ro** [comment]
Two questions @matthiasgeihs 
- I was not able to update to go-perun v0.8.0 because client.New then requires a watcher as 6th argument. Any idea on how to solve this problem [here](https://github.com/leandro-ro/perun-examples/blob/87856652ac18bfcfd1ac300c1f72fa58e5331745/payment-channel/client/setup.go#L104) ?
- How did you intend to use `c.PerunClient.StateChClient.OnNewChannel()` to replace `c.HandleNewChannel(ch)` [here](https://github.com/leandro-ro/perun-examples/blob/87856652ac18bfcfd1ac300c1f72fa58e5331745/payment-channel/client/handle.go#L39). I am not sure what the function given as argument must provide.

**Review by matthiasgeihs — COMMENTED** [review]
I like! Nice that you could find a structure for the payment channel example that can be easily extended for the app channel example.

Anyhow, since I had some ideas for improvements as well, instead of commenting here, I gave it a go and tried to further reduce the complexity of the code myself. You can find the result for the payment channel example at https://github.com/perun-network/perun-examples/tree/draft/leandro-review/payment-channel. My hope is that the app channel example can be based on that just as you showed for the previous version.

I suggest that you make yourself familiar with my suggested version. Feel free to comment! If it looks good, you could start revising the tutorial text accordingly. Note that there are still `//todo`'s in the code, which we should address one way or the other and then remove. Then, we should also try to adapt the app channel example accordingly.

**Comment by leandro-ro** [comment]
Please have a look at the app-channel code @matthiasgeihs . Thanks!

**Pull request closed without merging**